uniswap-v2-core/contracts/UniswapV2Factory.sol

```diff
@@ -37,13 +47,36 @@ contract UniswapV2Factory is IUniswapV2Factory {
         emit PairCreated(token0, token1, pair, allPairs.length);
     }
 
-    function setFeeTo(address _feeTo) external {
-        require(msg.sender == feeToSetter, 'UniswapV2: FORBIDDEN');
+    function setFeeTo(address _feeTo) external onlyOwner {
         feeTo = _feeTo;
     }
-
-    function setFeeToSetter(address _feeToSetter) external {
-        require(msg.sender == feeToSetter, 'UniswapV2: FORBIDDEN');
-        feeToSetter = _feeToSetter;
+    
+    function setDefaultSwapFee(uint _swapFee) external onlyOwner {
+        require(_swapFee < MAX_SWAP_FEE, "Vexchange: INVALID_SWAP_FEE");
+        require(_swapFee > MIN_SWAP_FEE, "Vexchange: INVALID_SWAP_FEE");
+        
+        defaultSwapFee = _swapFee;
+    }
+    
+    function setDefaultPlatformFee(uint _platformFee) external onlyOwner {
+        require(_platformFee < MAX_PLATFORM_FEE, "Vexchange: INVALID_PLATFORM_FEE");
+        
+        defaultPlatformFee = _platformFee;
+    }
+    
+    function setDefaultRecoverer(address _recoverer) external onlyOwner {
+        defaultRecoverer = _recoverer;
+    }
+    
+    function setSwapFeeForPair(address _pair, uint _swapFee) external onlyOwner {
+        IUniswapV2Pair(_pair).setSwapFee(_swapFee);
+    }
+    
+    function setPlatformFeeForPair(address _pair, uint _platformFee) external onlyOwner {
+        IUniswapV2Pair(_pair).setPlatformFee(_platformFee);
+    }
+    
+    function setRecovererForPair(address _pair, address _recoverer) external onlyOwner {
```

Comment:
No timelock on this?

Reply:
This function _SHOULD_ only be able to recover non-core assets (i.e. it should only allow us to withdraw assets that would have otherwise been stuck). If we want to make this timelocked, we can make the owner a timelock contract and thus timelock the management of the recoverer.

We could even make the owner a contract that hardcodes the recoverer for all time (if we're that confident we don't want to change it).

Reply:
You're right, missed it